src/Control/Carrier/Choose/Church.hs

```diff
@@ -1,4 +1,10 @@
 {-# LANGUAGE DeriveTraversable, FlexibleInstances, LambdaCase, MultiParamTypeClasses, RankNTypes, TypeOperators, UndecidableInstances #-}
+-- | Provides 'ChooseC', a carrier for 'Choose' effects (nondeterminism via choice).
+--
+-- It can be invoked with custom functions for choice, success, and failure
+ -- ('runChoose'), or it can delegate said operations to an 'S.Semigroup' instance
+ -- ('runChooseS'). Under the hood, it uses a Church-encoded structure and a binary
+ -- tree to prevent the problems associated with a naïve list-based implementation.
```

Comment:
1. Good catch, fixing.
2. At first, I disagreed, but maybe you’re right.
3. I agree it’s a good idea.
4. I tried to be descriptive without being too verbose. I like, for example, the docs for [`StateT`](http://hackage.haskell.org/package/transformers-0.5.6.2/docs/Control-Monad-Trans-State-Strict.html) from `transformers`: it provides a description of what it is, possible alternatives, and important caveats re. strictness. How about

```
{- |
A carrier for 'Choose' effects (nondeterminism via choice).

Under the hood, it uses a Church-encoded binary tree to avoid the problems associated with a naïve list-based implementation (see ["ListT done right"](http://wiki.haskell.org/ListT_done_right)).
-}
```